src/Common/ThreadProfileEvents.cpp

```diff
@@ -124,8 +124,12 @@ TasksStatsCounters::TasksStatsCounters(const UInt64 tid, const MetricsProvider p
                 };
         break;
     case MetricsProvider::Procfs:
-        stats_getter = [metrics_provider = std::make_shared<ProcfsMetricsProvider>(tid)]()
+        /// Note that in the case of Procfs we are always reading the same files over an over
+        /// In order to avoid opening and closing them for every task we use a ThreadLocal variable so we'll keep
+        /// the files under this thread until the thread exits
+        stats_getter = [tid]()
                 {
+                    thread_local auto metrics_provider = std::make_shared<ProcfsMetricsProvider>(tid);
```

Comment:
my only question is: is it possible to put this `thread_local` higher up the hierarchy (e.g. at `TasksStatsCounters` instance or even higher) to potentially save more initialisation costs